src/prelogin-payload.js

```diff
@@ -63,7 +64,8 @@ module.exports = class PreloginPayload {
       this.createEncryptionOption(),
       this.createInstanceOption(),
       this.createThreadIdOption(),
-      this.createMarsOption()
+      this.createMarsOption(),
+      ...(this.options.fedAuthRequested ? [this.createFedAuthOption()] : [])
```

Comment:
I'm trying to understand the need for spread operator here. `createFedAuthOption()` returns single object, so when is `...` put into use?

Reply:
Yeah, this code is a tiny bit convoluted. 😅 

What do you think about changing this to:

```js
    const options = [
      this.createVersionOption(),
      this.createEncryptionOption(),
      this.createInstanceOption(),
      this.createThreadIdOption(),
      this.createMarsOption()
    ];

    if (this.options.fedAuthRequested) {
      options.push(this.createFedAuthOption());
    }
```

I think this might be less confusing. 😄

Reply:
@v-suhame if fedauthreq is true, it spreads the first array which contains call to fedauth, otherwise the empty array, which nothing :metal:, as @arthurschreiber said it is convoluted, and I'll make the changes